### PR TITLE
Add refusalReasonCode and refusalReason to payments API response

### PR DESCRIPTION
**Pull request by RokPopov**
## Summary
Limiting AUTHORISE case response to only have the value of isFinal boolean and resultCode and adding CANCELLED case to the error/refused block.

## Tested scenarios
- Authorised payment
- Refused payment
- Error payment
- Cancelled payment

**Fixed issue**:  
[PW-6014]

### CODE REVIEW

File: src/Handlers/PaymentResponseHandler.php [hunk 1]
```diff
@@ -307,16 +307,23 @@ public function handleAdyenApis(
         $resultCode = $paymentResponseHandlerResult->getResultCode();
         $refusalReason = $paymentResponseHandlerResult->getRefusalReason();
         $refusalReasonCode= $paymentResponseHandlerResult->getRefusalReasonCode();
```
Comment (AndreasA) [inline src/Handlers/PaymentResponseHandler.php:309]:
```suggestion
        $refusalReasonCode = $paymentResponseHandlerResult->getRefusalReasonCode();
```

**Comment by AndreasA** [comment]
@RokPopov As mentioned in the issue this seems to intend fix the problem regarding cancelled and missing refusal code https://github.com/Adyen/adyen-shopware6/issues/209#issuecomment-1066923026
tested with 22, 23 and 25 I think. 22 was cancelled and 25 was refused.

**Review by peterojo — DISMISSED** [review]

**Review by acampos1916 — APPROVED** [review]

**Review by acampos1916 — APPROVED** [review]

**Review by peterojo — APPROVED** [review]

**Comment by acampos1916** [comment]
@RokPopov please rename the PR to a descriptive title so that the commit message is useful.

**Comment by AndreasA** [comment]
Just wondering what the state for this PR is?

**Comment by RokPopov** [comment]
Hey @AndreasA, the PR has been merged to develop branch. Kindly note we are also preparing a release soon.

**Pull request merged**